components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -776,9 +811,20 @@ private static void setMgwAPISecurityAndScopes(ExtendedAPI api, OpenAPI openAPI)
         return new String[]{securitySchemas, scopes};
     }
 
-    public static List<String> getMgwResourceSecurity(Operation operation) {
+    public static List<String> getMgwResourceSecurity(Operation operation, ApplicationSecurity appSecurity) {
+        Map<String, Object> operationExtensions = operation.getExtensions();
+        //override api level application security extension by operation level extension
+        if (operationExtensions != null && operationExtensions.containsKey(OpenAPIConstants.APPLICATION_SECURITY)) {
+            try {
+                Object appSecurityExtension = operationExtensions.get(OpenAPIConstants.APPLICATION_SECURITY);
+                appSecurity = new ObjectMapper().convertValue(appSecurityExtension, ApplicationSecurity.class);
```

Comment:
Can't we directly use the extension here rather than assigning it to a object and then passing it to objectmapper